plugin/pkg/auth/authorizer/rbac/bootstrappolicy/policy_test.go

```diff
@@ -192,19 +197,41 @@ func TestBootstrapClusterRoleBindings(t *testing.T) {
 }
 
 func TestBootstrapControllerRoles(t *testing.T) {
-	list := &api.List{}
-	names := sets.NewString()
-	roles := map[string]runtime.Object{}
-	bootstrapRoles := bootstrappolicy.ControllerRoles()
-	for i := range bootstrapRoles {
-		role := bootstrapRoles[i]
-		names.Insert(role.Name)
-		roles[role.Name] = &role
+	tests := map[string]struct {
+		filename        string
+		enabledFeatures []featuregate.Feature
+	}{
+		"with feature gates enabled3": {
+			enabledFeatures: []featuregate.Feature{features.PodDisruptionConditions,
+				features.StatefulSetAutoDeletePVC,
+				genericfeatures.StorageVersionAPI,
+				genericfeatures.APIServerIdentity},
```

Comment:
why do these features matter?

Reply:
refactored the test to only focus on the problematic feature gate.